sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/internal/changefeed/implementation/PartitionProcessorImpl.java

```diff
@@ -157,13 +163,21 @@ public PartitionProcessorImpl(ChangeFeedObserver observer, ChangeFeedContextClie
                             }
                         }
                         default: {
-                            this.logger.error("Unrecognized DocDbError enum value {}", docDbError, clientException);
+                            logger.error("Unrecognized Cosmos exception returned error code {}", docDbError, clientException);
                             this.resultException = new RuntimeException(clientException);
                         }
                     }
+                } else if (throwable instanceof LeaseLostException) {
+                    logger.info("LeaseLoseException with partition {} from thread {}",
+                        this.options.partitionKey().getInternalPartitionKey(), Thread.currentThread().getId());
+                    this.resultException = (LeaseLostException) throwable;
                 } else if (throwable instanceof TaskCancelledException) {
-                    this.logger.debug("Exception: partition {}", this.settings.getPartitionKeyRangeId(), throwable);
+                    logger.debug("Task cancelled exception: partition {} from {}",
+                        this.settings.getPartitionKeyRangeId(), Thread.currentThread().getId(), throwable);
                     this.resultException = (TaskCancelledException) throwable;
+                } else {
+                    logger.warn("Unexpected exception from thread {}", Thread.currentThread().getId(), throwable);
+                    this.resultException = new RuntimeException(throwable);
```

Comment:
just please be aware that this doesn't record exception stacktrace. If that matters. this will maintain the original exception stacktrace though.

Is that ok?

Reply:
yes; the reason we create a RuntimeException here and assign it to the resultException is to signal to the "parent" that the processing worker should be killed along with the corresponding renewal worker.